### PR TITLE
GUI: add option aspect ratio

### PR DESCRIPTION
**Pull request by raven02**
Add new option aspect ratio 
Untitled

### CODE REVIEW

File: src/yuzu/configuration/configure_graphics.cpp [hunk 1]
```diff
@@ -24,6 +24,30 @@ enum class Resolution : int {
     Scale4x,
 };
 
+enum class Aspect : int {
+    ar16_9,
```
Comment (DarkLordZach) [inline src/yuzu/configuration/configure_graphics.cpp]:
```suggestion
    AR16_9,
```

File: src/yuzu/configuration/configure_graphics.cpp [hunk 2]
```diff
@@ -24,6 +24,30 @@ enum class Resolution : int {
...
+    ar21_9,
```
Comment (DarkLordZach) [inline src/yuzu/configuration/configure_graphics.cpp]:
```suggestion
    AR21_9,
```

File: src/yuzu/configuration/configure_graphics.cpp [hunk 3]
```diff
@@ -24,6 +24,30 @@ enum class Resolution : int {
     Scale4x,
 };
 
+enum class Aspect : int {
```
Comment (DarkLordZach) [inline src/yuzu/configuration/configure_graphics.cpp]:
```suggestion
enum class AspectRatio {
```
int is unnecessary

File: src/yuzu/configuration/configure_graphics.cpp [hunk 4]
```diff
@@ -24,6 +24,30 @@ enum class Resolution : int {
...
+    ar16_9,
+    ar21_9,
+};
+
+float ToAspectRatio(Aspect option) {
```
Comment (DarkLordZach) [inline src/yuzu/configuration/configure_graphics.cpp]: Remove this function -- it doesnt do what it says and should be inlined

File: src/yuzu/configuration/configure_graphics.cpp [hunk 5]
```diff
@@ -24,6 +24,30 @@ enum class Resolution : int {
...
+    switch (option) {
+    case Aspect::ar16_9:
+        return 0.f;
+    case Aspect::ar21_9:
+        return 1.f;
+    }
+    return 0.f;
+}
+
+Aspect FromAspectRatio(float aspect) {
```
Comment (DarkLordZach) [inline src/yuzu/configuration/configure_graphics.cpp]: Remove this function -- it doesnt do what it says and should be inlined

File: src/yuzu/configuration/configure_graphics.ui [hunk 6]
```diff
@@ -138,6 +138,33 @@
...
+            <property name="text">
+             <string>Aspect Ratio:</string>
+            </property>
+           </widget>
+          </item>
+          <item>
+           <widget class="QComboBox" name="aspect_ratio_combobox">
+            <item>
+             <property name="text">
+              <string>Keep aspect ratio 16:9</string>
```
Comment (DarkLordZach) [inline src/yuzu/configuration/configure_graphics.ui]:
```suggestion
              <string>Default (16:9)</string>
```

File: src/core/frontend/framebuffer_layout.cpp [hunk 7]
```diff
@@ -27,8 +27,10 @@ FramebufferLayout DefaultFrameLayout(u32 width, u32 height) {
     // so just calculate them both even if the other isn't showing.
     FramebufferLayout res{width, height};
 
-    const float emulation_aspect_ratio{static_cast<float>(ScreenUndocked::Height) /
-                                       ScreenUndocked::Width};
+    const float emulation_aspect_ratio =
```
Comment (DarkLordZach) [inline src/core/frontend/framebuffer_layout.cpp]: This doesn't force 21:9 if 21:9 is selected -- it just matches the user's screen
If I am in 16/9 and select 21/9 it should work & not just keep 16/9

File: src/core/settings.h [hunk 8]
```diff
@@ -429,6 +429,7 @@ struct Values {
     int vulkan_device;
 
     float resolution_factor;
+    float aspect_ratio;
```
Comment (DarkLordZach) [inline src/core/settings.h]: Make this an enum (use the enum from elsewhere)

**Comment by raven02** [comment]
Since 21:9 need specific mod per game , i keep it as 'stretch to window' and 'default (16:9)'

**Comment by raven02** [comment]
I am closing this as another contributor has better idea

**Comment by Margen67** [comment]
Superseded by #3412.

**Pull request closed without merging**